src/integration-test/java/com/commercetools/sync/integration/commons/utils/StateITUtils.java

```diff
@@ -35,28 +60,106 @@ public static void deleteStatesFromTargetAndSourceByType() {
      *
      * @param ctpClient defines the CTP project to delete the states from.
      */
-    public static void deleteStates(@Nonnull final SphereClient ctpClient) {
-        queryAndExecute(ctpClient, StateQueryBuilder
+    public static <T extends State> void deleteStates(@Nonnull final SphereClient ctpClient) {
+        //delete transitions
+        QueryExecutionUtils.queryAll(ctpClient, StateQueryBuilder
             .of()
-            .plusPredicates(QueryPredicate.of("builtIn = false"))
-            .build(),
-            StateDeleteCommand::of);
+            .plusPredicates(QueryPredicate.of("builtIn = false")).build())
+            .thenCompose(result -> {
+                final List<CompletionStage<State>> clearStates = new ArrayList<>();
+                result.stream().forEach(state -> {
+                    if (state.getTransitions() != null && !state.getTransitions().isEmpty()) {
+                        List<? extends UpdateAction<State>> emptyUpdateActions = Collections.emptyList();
+                        clearStates.add(ctpClient.execute(StateUpdateCommand.of(state, emptyUpdateActions)));
+                    } else {
+                        clearStates.add(CompletableFuture.completedFuture(state));
+                    }
+                });
+                return listOfFuturesToFutureOfList(clearStates);
+            }).thenAccept(clearedStates -> clearedStates.forEach(stateToRemove ->
+            ctpClient.execute(StateDeleteCommand.of(stateToRemove)))
+        ).handle((result, throwable) -> {
+            if (throwable != null) {
```

Comment:
This looks like unnecessary, it should throw the error.

Reply:
8ba612fa